src/afl-fuzz-init.c

```diff
@@ -710,7 +710,10 @@ void read_testcases(afl_state_t *afl, u8 *directory) {
 
   }
 
-  for (i = 0; i < (u32)nl_cnt; ++i) {
+  i = nl_cnt;
+  do {
```

Comment:
why not keep this as (more readable) for loop? What happens if `nl_cnt == 0`?

Reply:
value 0 is already handled above. I found reversing the for loop less readable and more error prone than moving to while.

Reply:
I don't think it's handled? If you'd use a for loop here we wouldn't have this discussion ;)

Reply:
At least make it a while loop instead of do while?

Reply:
> I don't think it's handled? If you'd use a for loop here we wouldn't have this discussion ;)

above, not visible in the diff is `if nl_cnt == 0) { .... return;}`.



> At least make it a while loop instead of do while?

I find it best like this, for the same reason I did not use for() is why I did not use while(), but do {} while instead.
change it if you want. but dont introduce a bug :)

Reply:
Where? All I see is`nl_cnt = scandir(dir, &nl, NULL, alphasort);`.
It's unlikely that it is `0` on linux as it should(?) return `.` and `..`,  so I guess it usually works(?), just not intuitive to look at :)

Reply:
uh right! I looked at foreign_sync. but it is impossible that it is empty, at least on unix, as all unix directories have "." and "..", so nl_cnt is always < 0 on error or at minimum 2.

Reply:
but for windows that is likely different so I just added a check. just not to run into this in the future.